### ISSUE TITLE
[Feature Requests] allow non-root user run rootless docker within rooted docker daemon.

### ISSUE THREAD

**Issue by HuJK** [moby/moby#41258]
I am a server admin, and I want to allow everyone on my server able to run docker command.

If I add everyone to docker group, it will grant every root permission because it's root equivalent.
Every user can change root's password with following command:
```
$ docker run -it -v /:/rootfs ubuntu bash
# chroot /rootfs
# passwd
```

If I ask everyone to use rootless docker, that will cause multiple docker engine running in the server. It's not a problem. 
The problem is same image can't share among different account, it will cause waste of disk space.

So I request a feature that a new socket named rootless_docker.sock which can pull/delete image, manage containers which create by himself , run image in rootless mode, bind non-privileged port.

Because the feature of rootless docker, userns-remap will prevent user modify sensitive files in host filesystem:
```
$ docker run -it -v /:/rootfs ubuntu bash
# chroot /rootfs
# passwd
Permission Denied
```

**Comment by thaJeztah** [comment]
I don't think this will be possible. The data store in `/var/lib/docker` is designed to be managed by a single daemon. The daemon keeps track if images/files/volumes are used, and based on that knows if they can be removed.

If multiple daemons (irregardless if they're "rootless" or "rootfull" share the same storage, then they won't know if images/files/volumes are in use by other daemons. As a result, "daemon A" would be able to remove images that are in use by "daemon B" and vice-versa.

_possibly_ it's possible (to an extend) if we integrate with the containerd image store (and making use of namespaces), but that would have to be looked at once we arrive there

**Comment by HuJK** [comment]
You're right......

Is it possible that only one rooted daemon, but it will spawn a rootless container(userns-remaped, cgroups disabled etc.) for non-root caller?
